components/brave_shields/resources/panel/components/tree-list/index.tsx

```diff
@@ -46,42 +53,100 @@ function groupByOrigin (data: Url[]) {
 
 function TreeList (props: Props) {
   const { siteBlockInfo, setViewType } = React.useContext(DataContext)
-  const mappedData = React.useMemo(() => groupByOrigin(props.data), [props.data])
+  const allowedList = props.allowedList ?
+    props.allowedList : []
+
+  const allowedScriptsByOrigin = React.useMemo(() =>
+    groupByOrigin(allowedList), [allowedList])
+
+  const blockedScriptsByOrigin = React.useMemo(() =>
+    groupByOrigin(props.blockedList),
+                  [props.blockedList])
+
+  const handleAllowAllScripts = () => {
+    const origins = props.blockedList.map(entry => new URL(entry.url).origin)
+    getPanelBrowserAPI().dataHandler.allowScriptsOnce(origins)
+  }
+
+  const handleBlockAllScripts = () => {
+    const origins = allowedList.map(entry => new URL(entry.url).origin)
+    getPanelBrowserAPI().dataHandler.blockAllowedScripts(origins)
+  }
+
+  const handleBlockScript = (name: string) => {
+    getPanelBrowserAPI().dataHandler.blockAllowedScripts([new URL(name).origin])
+  }
+
+  const handleAllowScript = !props.allowedSectionVisible ? undefined
+    : (name: string) => {
+      getPanelBrowserAPI().dataHandler.allowScriptsOnce([new URL(name).origin])
+    }
 
   return (
     <S.Box>
       <S.HeaderBox>
         <S.SiteTitleBox>
           <S.FavIconBox>
-            <img key={siteBlockInfo?.faviconUrl.url} src={siteBlockInfo?.faviconUrl.url} />
+            <img src={siteBlockInfo?.faviconUrl.url} />
           </S.FavIconBox>
           <S.SiteTitle>{siteBlockInfo?.host}</S.SiteTitle>
         </S.SiteTitleBox>
-        <S.Grid>
-          <span>{props.totalBlockedCount}</span>
-          <span>{props.blockedCountTitle}</span>
-        </S.Grid>
       </S.HeaderBox>
-      <S.TreeBox>
+      <S.Scroller>
+        {props.allowedSectionVisible && (
         <div>
```

Comment:
Do we need another div here or can we get away with Fragment `<> </>`?

Reply:
replaced by `<> </>`